packages/office-ui-fabric-react/src/components/pickers/AutoFill/BaseAutoFill.tsx

```diff
@@ -154,37 +167,71 @@ export class BaseAutoFill extends BaseComponent<IBaseAutoFillProps, IBaseAutoFil
       this.props.onKeyDown(ev);
     }
 
-    switch (ev.which) {
-      case KeyCodes.backspace:
-        this._autoFillEnabled = false;
-        break;
-      case KeyCodes.left:
-        if (this._autoFillEnabled) {
+    // If the event is actively being composed, then don't alert autofill.
+    // Right now typing does not have isComposing, once that has been fixed any should be removed.
+    if (!(ev.nativeEvent as any).isComposing) {
+      switch (ev.which) {
+        case KeyCodes.backspace:
           this._autoFillEnabled = false;
-        }
-        break;
-      case KeyCodes.right:
-        if (this._autoFillEnabled) {
-          this._autoFillEnabled = false;
-        }
-        break;
-      default:
-        if (!this._autoFillEnabled) {
-          if (this.props.enableAutoFillOnKeyPress!.indexOf(ev.which) !== -1) {
-            this._autoFillEnabled = true;
+          break;
+        case KeyCodes.left:
+          if (this._autoFillEnabled) {
+            this._value = this.state.displayValue!;
+            this._autoFillEnabled = false;
           }
-        }
-        break;
+          break;
+        case KeyCodes.right:
+          if (this._autoFillEnabled) {
+            this._value = this.state.displayValue!;
+            this._autoFillEnabled = false;
+          }
+          break;
+        default:
+          if (!this._autoFillEnabled) {
+            if (this.props.enableAutoFillOnKeyPress!.indexOf(ev.which) !== -1) {
```

Comment:
I'm seeing a lot of this.props.foo!
Remember that Partial<T> exists now. Maybe you can make an interface of IPropsAllRequired, and expose Partial<IPropsAllRequired> as the actual props the component takes, and then use IPropsAllRequired internally.